client/protocols/rate_limit.py

```diff
@@ -4,20 +4,73 @@
 import time
 import textwrap
 
+class Hint:
+    def __init__(self, text='', cooldown=0):
+        self.text = text
+        self.cooldown = cooldown
 
-COOLDOWN_POLICY = (0, 60,) # uniform 60s cooldown
+Hint.NONE = Hint()
 
+EXAMPLES = \
+"""Try explaining the problem, its domain (inputs), range (outputs),
+and behavior in your own words. Come up with a few example inputs
+and outputs.
+"""
 
-COOLDOWN_MSG = \
+PROCESS = \
+"""Try identifying where you are in the problem-solving process
+
+    https://cs61a.org/articles/studying.html#solving-problems
 """
-You're spamming the autograder for "{question}"!
-Please wait {wait}s... (attempts so far: {tries})
 
-If you're stuck on "{question}", try talking to your neighbor,
-asking for help, or running your code in interactive mode:
+TEST = \
+"""Try testing a few of your own examples in the interpreter:
 
     python3 -i {files}
 
+You can also run the doctests:
+
+    python3 -m doctest {files}
+"""
+
+PYTHON_TUTOR = \
+"""Try plugging the code along with an example into Python Tutor
+to visualize Python's process with
+
+    http://tutor.cs61a.org
+"""
+
+DIAGRAM = \
+"""Try walking through your approach with a friend and drawing
+your solution on the board. Verify that the strategy matches
+with the implementation.
+"""
+
```

Comment:
I feel like the `DIAGRAM` hint as it's worded now is only relevant for labs and hws/projects done in OH, but not so much for assignments done individually at home.